GoogleForCreators/web-stories-wp
# Settings: Allow disabling archive page

## Context

See #4862 for history / context on this matter.

## Summary

We want to allow users to disable the default /web-stories archive page.

## Relevant Technical Choices

As there are only two options for now, enabled and disabled, made this PR simpler and just made the dashboard setting a checkbox. If we have more options in the future, at that point change to a dropdown.

## To-do



## User-facing changes

#### New setting in dashboard
<img width="1001" alt="Screenshot 2021-09-09 at 13 43 25" src="https://user-images.githubusercontent.com/237508/132687954-2cb02cc9-14ee-4a54-a1b0-08497cc3df61.png">
<img width="1010" alt="Screenshot 2021-09-09 at 13 43 35" src="https://user-images.githubusercontent.com/237508/132687976-6b7d4478-5ffe-4aab-bb07-baf60af4a184.png">


#### Archive link option does not show for blocks
<img width="307" alt="Screenshot 2021-09-06 at 17 04 35" src="https://user-images.githubusercontent.com/237508/132242799-225a3d09-dfb9-4cc6-ab8f-00d26b89b4f2.png">
<img width="313" alt="Screenshot 2021-09-06 at 17 04 47" src="https://user-images.githubusercontent.com/237508/132242860-0caf6e33-da8f-46de-84e9-b01547c3fc57.png">
<img width="740" alt="Screenshot 2021-09-06 at 17 04 55" src="https://user-images.githubusercontent.com/237508/132242874-7a909d9c-c762-47f9-a726-ed490ccb3ca0.png">

#### When archive is disabled, link should not rendered on the front end
<img width="734" alt="Screenshot 2021-09-06 at 17 05 07" src="https://user-images.githubusercontent.com/237508/132242923-91e40949-a5c3-4946-9eeb-1cc63c0453af.png">

#### Options should render in widget form to show archive links
<img width="696" alt="Screenshot 2021-09-06 at 17 08 36" src="https://user-images.githubusercontent.com/237508/132243171-7e1f9741-80f6-4b96-8921-fed797c71b77.png">


#### Archive url, should 404.
<img width="1904" alt="Screenshot 2021-09-06 at 17 05 28" src="https://user-images.githubusercontent.com/237508/132243000-b85455a7-6d3e-4f63-afbe-865341eef408.png">
 


## Testing Instructions






This PR can be tested by following these steps:

**Note:** Ensure the "Disable Archive" experiment is enabled first. 

#### Test dashboard setting
1. Go to dashboard > settings. 
2. Ensure Stories Archive option is present. 
3. Disable / Enable archives, reload page, ensure setting is saved. 

#### Test archive is really disabled
1. Diable Stories Archive ( see above ). 
2. Go to /web-stories on the front end. 
3. Ensure page is a 404. 

#### Test widget presents settings are not rendered
1. Diable Stories Archive ( see above ). 
2. Enable classic widget plugin. 
3. Create a web stories widget. 
4. Ensure no options for archives are present. 
5. Save widget. 
6. Ensure no link archive is rendered on front end.

#### Test block settings are not rendered
1. Diable Stories Archive ( see above ). 
2. Create a new post
3. Create a web stories block
4. Select Latest stories.
5. Select any layout style
6. Ensure no options for archives are present. 
7. Save post. 
8. Ensure no link archive is rendered on front end.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8811
